halo2_backend/src/plonk/prover.rs

```diff
@@ -114,16 +116,22 @@ pub struct ProverV2<
 > {
     // Circuit and setup fields
     params: &'params Scheme::ParamsProver,
+    // Plonk proving key
     pk: &'a ProvingKey<Scheme::Curve>,
-    // TODO: Add getter
-    pub phases: Vec<sealed::Phase>,
-    // State
-    instance: Vec<InstanceSingle<Scheme::Curve>>,
-    advice: Vec<AdviceSingle<Scheme::Curve, LagrangeCoeff>>,
+    // Phases
+    phases: Vec<sealed::Phase>,
+    // Polynomials (Lagrange and Coeff) for all circuits instances
+    instances: Vec<InstanceSingle<Scheme::Curve>>,
+    // Advice polynomials with its bindings
+    advices: Vec<AdviceSingle<Scheme::Curve, LagrangeCoeff>>,
+    // The challanges for each phase
```

Comment:
The map indexes challenges by challenge index, and a phase may have several challenges.
Since this map doesn't index challenges by phase, I think it's more clear if the description is:
`The phase challenges by challenge index`